ui/src/app/components/help-sidebar.tsx

```diff
@@ -272,6 +300,65 @@ export const HelpSidebar = withUserProfile()(
       return highlightSearchTerm(searchTerm, content, colors.success);
     }
 
+    renderWorkspaceMenu() {
+      const {deleteFunction, shareFunction, workspace, workspace: {accessLevel, id, namespace}} = this.props;
+      const isNotOwner = !workspace || accessLevel !== WorkspaceAccessLevel.OWNER;
+      return <PopupTrigger
+        side='bottom'
+        closeOnClick
+        content={
+          <React.Fragment>
+            <div style={styles.dropdownHeader}>Workspace Actions</div>
+            <MenuItem
+              icon='copy'
+              onClick={() => {
+                AnalyticsTracker.Workspaces.OpenDuplicatePage();
+                NavStore.navigate(['/workspaces', namespace, id, 'duplicate']);
+              }}>
+              Duplicate
+            </MenuItem>
+            <MenuItem
+              icon='pencil'
+              tooltip={isNotOwner && 'Requires owner permission'}
+              disabled={isNotOwner}
+              onClick={() => {
+                AnalyticsTracker.Workspaces.OpenEditPage();
+                NavStore.navigate(['/workspaces', namespace, id, 'edit']);
+              }}>
+              Edit
+            </MenuItem>
+            <MenuItem
+              icon='share'
+              tooltip={isNotOwner && 'Requires owner permission'}
+              disabled={isNotOwner}
+              onClick={() => {
+                AnalyticsTracker.Workspaces.OpenShareModal();
+                shareFunction();
+              }}>
+              Share
+            </MenuItem>
+            <MenuItem
+              icon='trash'
+              tooltip={isNotOwner && 'Requires owner permission'}
```

Comment:
opt nit: can this be put in a function since its been repeated in every menuitem?

Reply:
Good idea, moved it to a const within render.